de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/RunnerNodeInternalSettings.java

```diff
@@ -37,40 +42,64 @@
    */
   public File imageFile = null;
 
-  public Image plot = null;
 
+/**
+ * TODO: Check if saving the image in a separate file is still necessary.
+ * */
   public RunnerNodeInternalSettings() {
     try {
+    
       imageFile = FileUtil.createTempFile("FskxRunner-", ".svg");
     } catch (IOException e) {
       LOGGER.error("Cannot create temporary file.", e);
       throw new RuntimeException(e);
     }
   }
 
-  /** Loads the saved image. */
-  public void loadInternals(File nodeInternDir) throws IOException {
-    final File file = new File(nodeInternDir, "Rplot.svg");
 
-    if (file.exists() && file.canRead()) {
-      FileUtil.copy(file, imageFile);
-      try (InputStream is = new FileInputStream(imageFile)) {
-        plot = new PNGImageContent(is).getImage();
-      }
-    }
-  }
+  /** save generated resource files in the /internal folder*/
+  public void saveInternals(File nodeInternDir, FskPortObject fskObj) throws IOException {
 
-  /** Saves the saved image. */
-  public void saveInternals(File nodeInternDir) throws IOException {
-    if (plot != null) {
-      final File file = new File(nodeInternDir, "Rplot.svg");
-      FileUtil.copy(imageFile, file);
-    }
-  }
+    if(fskObj instanceof CombinedFskPortObject) {
+
+      FskPortObject firstObj    = ((CombinedFskPortObject)fskObj).getFirstFskPortObject();
+      FskPortObject secondObj   = ((CombinedFskPortObject)fskObj).getSecondFskPortObject();
+
+      saveInternals(nodeInternDir,firstObj);
+      saveInternals(nodeInternDir,secondObj);
+
+    } else {
+
+      // FSK Object contains paths to resource files (if there are any).
+      // On saving thie workflow, these files are copied to the /internal 
+      // folder of the Runner node that created the files
+      List<File> temp_dirs = new ArrayList<File>(); 
+      for(Path path : fskObj.generatedResourceFiles.getResourcePaths()) {
+
+        File resourceFile = path.toFile();
+        
+        temp_dirs.add(resourceFile.getParentFile());
+
+        final File internalFile = new File(nodeInternDir, resourceFile.getName());
+
+        FileUtil.copy(resourceFile, internalFile);
+        
+      }//for
+
+      // Directories with temporary files are deleted since their copies are in the /internal folder.
+      for(File dir : temp_dirs) {
+        try {
+          FileUtil.deleteRecursively(dir);  

```

Comment:
If you are not checking the exception I would replace this with  [FileUtils#deleteQuietly](https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FileUtils.html#deleteQuietly(java.io.File)). It does not throw an exception.